### ISSUE TITLE
What is the class id for background class?

### ISSUE THREAD

**Issue by YutingZhang** [facebookresearch/detr#32]
I have a quick question: is the background class id 0 or 91? (DETR used 91 COCO categories to train)
It seems the `targets` object returned by dataloader uses 1-91 for all the object categories, but the `loss_labels` function used 91 instead of 0 for background. I am not sure if I missed something. 

Thanks.

**Comment by fmassa** [comment]
Hi,

No-object label is `num_classes + 1` in DETR, so the last element in the prediction.

There is no need for the dataset to return no-object classes (as it return only existing objects). The only place where the notion of the no-object position is taken into account  in the loss and postprocessor.

Let us know if you have further questions

**Comment by raviv** [comment]
@fmassa usually when training 10 classes, 0 is for bacgkround and 1-10 are actual classes.
Do you mean that here 0-9 should be the classes and 10 is for background?
Also, when constructing the DETR class should num_classes param be 10 or 11?
Thanks.

**Comment by fmassa** [comment]
@raviv DETR uses the last value for the class.
So if you have 10 classes, you should set `num_classes` to 11, and class index 10 will be the background,

**Comment by hariSky** [comment]
@fmassa 
I am bit confused with your answer after the EDIT above. Does the first statement hold?
"So if you have 10 classes, you should set num_classes to 10 and class index 10 will be the background,"

**Comment by qiaoran-dawnlight** [comment]
@hariSky Yes, the mode build with real num_class N that no need for background class. But the model itself set with N+1. The N+1 is no object.

**Comment by zhishao** [comment]
@fmassa [fmassa](https://github.com/fmassa) Hi , can DETR train images that without any object on it ? I mean the negtive data .

**Comment by rsong0606** [comment]
@fmassa same question as @zhishao . For an image without an object(negative example), how do we label it in COCO format? Do we label the annotations bbox as all zeros? Or we do not even label it, since there's an extra class built in to handle no-object? Thanks!

**Comment by frabob2017** [comment]
I want to ask the same question, I want to compare YOLO and DETR. I found that in YOLO, an image without an object(background images) just leave it alone. YOLO will take care of it. Does DETR do the same way, for the background images?

**Comment by vritansh** [comment]
I wanted to know if there's a clarity on this. I've total of 3 object classes, Since DETR outputs num_classes + 1. In coco format how should I even label the negative samples, should I keep the category_id=4 and for bbox=[0,0,0,0].  Any help would be greatly appreciated. Thanks in advance !
